### ISSUE TITLE
List of individual pipelines not always displayed in OSIO space dashboard

### ISSUE THREAD

**Issue by ldimaggi** [openshiftio/openshift.io#3461]
Steps to recreate:
* Create a new space
* Create a new app/quickstart
* Navigate from the pipeline page: https://openshift.io/username/spacename/create/pipelines to the space dashboard page by going to account home and then the recent space page
* Most of the time, the pipeline for the recently created app is not displayed.

For example:

The pipeline is here at 10:51:05AM:

![screenshot from 2018-05-04 10-51-05](https://user-images.githubusercontent.com/642621/39636469-4c40df76-4f8e-11e8-988b-ac9054683cf5.png)

And - now it's gone at 10:51:21AM:

![screenshot from 2018-05-04 10-51-21](https://user-images.githubusercontent.com/642621/39636470-4c555ed8-4f8e-11e8-8fb6-2e56adf728cd.png)

And - now it's back at 10:58:21AM:

![screenshot from 2018-05-04 10-58-21](https://user-images.githubusercontent.com/642621/39636473-4c88e3f2-4f8e-11e8-897a-7711e59fdf78.png)

**Comment by ldimaggi** [comment]
Issue found/encountered by EE tests. This is causing what appear to be random test failures.

**Comment by jiekang** [comment]
@ldimaggi in your experience, it only appears/disappears when visiting the page right? If you sit on the page and don't navigate elsewhere for some length of time, does it switch states?

**Comment by ldimaggi** [comment]
Not sure - the problem that I logged seems to have become non-reproducible. Has something changed?

**Comment by jiekang** [comment]
I think it may have been exacerbated by https://github.com/openshiftio/openshift.io/issues/3380 (huge number of http requests being made on the page that slowed everything on it down) which has been fixed.

I can still reproduce on my local instance by visiting Analyze -> Create -> Analyze. Sometimes the pipeline list will show, other times it won't; it looks like it occurs at a much lower rate. I just wanted to make sure the list never disappears when just sitting on the page as I haven't experienced that.

**Comment by ldimaggi** [comment]
Still seeing this randomly.

**Comment by jiekang** [comment]
Yes, I still see it in my local instance as well. The PR is available.

**Comment by ldimaggi** [comment]
![screenshot from 2018-05-08 10-32-41](https://user-images.githubusercontent.com/642621/39763498-461042de-52ab-11e8-8854-58452dcebcf2.png)

**Comment by Preeticp** [comment]
@ldimaggi I am not sure this is the same problem. We are updating the docs as per latest changes on the UX. I am on experimental view and I just created an app following the launcher, when I go to Analyze/Dashboard page, I do not see a section for Pipelines at all. I tried resetting my environment and then created a fresh App, and this what I see:

![screenshot from 2018-05-08 21-03-16](https://user-images.githubusercontent.com/16647625/39767469-7a841a6e-5304-11e8-980b-eb6beea8a569.png)
I am trying to see if we need to remove references in our docs to Pipelines in the dashboard. Any input would be welcome.

**Comment by jiekang** [comment]
In the new dashboard, the Pipelines widget no longer exists. I think this is expected. @joshuawilson ?

**Comment by Preeticp** [comment]
Thanks @jiekang, then the docs team is on the right track. We are removing all references in the docs to Pipelines in the Dashboard view.

**Comment by joshuawilson** [comment]
@Preeticp but that is the experimental view. You should only be pre-loading the Beta view.

**Comment by Preeticp** [comment]
Got it, thanks @joshuawilson

**Comment by jiekang** [comment]
I was planning to try to find the root cause for this but I can't reproduce it on production or on a local instance anymore. I wonder what changed for me... :(

**Comment by ldimaggi** [comment]
Followup on Pavol's earlier comment - this problem seems to be seen much more often on the 1a and 1b clusters (less often on 2 and 2a).  Maybe timing related?

**Comment by jiekang** [comment]
I think it's certain there is a race, I just don't know where. The pipelines observables are a bit strange. They emit a lot of different values sometimes.

**Comment by ldimaggi** [comment]
It looks as though issue #3461 is being seen 100% of the time for user accounts on the 1a and 1b clusters - See: #3527

**Comment by aptmac** [comment]
fwiw I encounter this infrequently, but when I do I'm usually able to recreate it after a bit of spam refreshing. I added a console.warn() in a local run to the `pipelines-widget.component.ts` to see what the `buildConfigs` observable [[0]](https://github.com/fabric8-ui/fabric8-ui/blob/c62508f72fd8a2475b2251a2aa9dfad518ada84e/src/app/dashboard-widgets/pipelines-widget/pipelines-widget.component.ts#L34) looked like when the widget is empty, and it has the info it just looks like the html isn't getting updated.

![pipelines-empty](https://user-images.githubusercontent.com/10425301/39827642-926c4cb4-5386-11e8-8480-54dfba0dce3f.png)

[0] https://github.com/fabric8-ui/fabric8-ui/blob/c62508f72fd8a2475b2251a2aa9dfad518ada84e/src/app/dashboard-widgets/pipelines-widget/pipelines-widget.component.ts#L34

**Comment by jiekang** [comment]
The DOM not being updated is what the PR targets and resolves that. Though I'm worried it could be a sign of a deeper underlying issue, this widget is going to be removed soon anyways.

**Comment by ljelinkova** [comment]
Is there any progress on this issue? It causes the E2E tests fail constantly (and no, we are not commenting out any part of the E2E test unless it is fixed).

**Comment by ldimaggi** [comment]
When will this fix be available in production? It's still broken there:

![screenshot from 2018-05-15 12-56-51](https://user-images.githubusercontent.com/642621/40071752-bace4654-583f-11e8-977e-805871bf2c0b.png)

**Comment by joshuawilson** [comment]
validating, then I'll update prod
